### ISSUE TITLE
SSL configuration

### ISSUE THREAD

**Issue by shajji1** [flutter/flutter#41602]
Please add **SSL** configuration like in **npm** _ssl-strict_. i can't access https://pub.dartlang.org/api/packages/cupertino_icons or any other package while running command **flutter packages get** in project directory. but when i access via browser it's letting me smoothly.

**Comment by shajji1** [comment]
i'm getting this error **Got TLS error trying to find package cupertino_icons at https://pub.dartlang.org.** while running command **flutter packages get**.
 
and getting following error detail, when running with `-v` flag.

> Get versions from https://pub.dartlang.org/api/packages/cupertino_icons.
> HTTP GET https://pub.dartlang.org/api/packages/cupertino_icons
> Accept: application/vnd.pub.v2+json
> X-Pub-OS: windows
> X-Pub-Command: get
> X-Pub-Session-ID: DB3C61BA-B4F9-42E9-8E06-27374A71BF49
> X-Pub-Environment: flutter_cli:get
> X-Pub-Reason: direct
> user-agent: Dart pub 2.5.0
> HTTP error:
> HandshakeException: Handshake error in client (OS Error:
>        CERTIFICATE_VERIFY_FAILED: self signed certificate in certificate chain(handshake.cc:352))

**Comment by escamoteur** [comment]
@shajji1 do you have a special network environment?

**Comment by Hixie** [comment]
@shajji1 Please include your `flutter doctor` output. What is the output of `SET` at the command line?

It sounds like there is something weird in your environment, like a proxy or something.

**Comment by shajji1** [comment]
following is **flutter doctor** output.

> [√] Flutter (Channel stable, v1.9.1+hotfix.2, on Microsoft Windows [Version 10.0.18362.356], locale en-US)
>     • Flutter version 1.9.1+hotfix.2 at C:\Flutter_v1.9.1_hotfix.2_stable
>     • Framework revision 2d2a1ffec9 (3 weeks ago), 2019-09-06 18:39:49 -0700
>     • Engine revision b863200c37
>     • Dart version 2.5.0
> 
> 
> [√] Android toolchain - develop for Android devices (Android SDK version 29.0.2)
>     • Android SDK at C:\ANDROID
>     • Android NDK location not configured (optional; useful for native profiling support)
>     • Platform android-29, build-tools 29.0.2
>     • ANDROID_HOME = C:\ANDROID
>     • Java binary at: C:\Program Files\Java\jdk1.8.0_221\bin\java
>     • Java version Java(TM) SE Runtime Environment (build 1.8.0_221-b11)
>     • All Android licenses accepted.
> 
> [!] Android Studio (not installed)
>     • Android Studio not found; download from https://developer.android.com/studio/index.html
>       (or visit https://flutter.dev/setup/#android-setup for detailed instructions).
> 
> [√] VS Code (version 1.38.1)
>     • VS Code at C:\Users\Shujahuddin\AppData\Local\Programs\Microsoft VS Code
>     • Flutter extension version 3.4.1
> 
> [!] Connected device
>     • Device TWMBE6NR69QOMROF is not authorized.
>       You might need to check your device for an authorization dialog.
> 
> ! Doctor found issues in 2 categories.

**Comment by shajji1** [comment]
@escamoteur yeah i'm working in my office, but 3 or 4 months ago i create flutter env easily, but now i install windows, so i  create flutter env again, but now i,m gettting error.

**Comment by shajji1** [comment]
@Hixie, following is **SET** command ouput

> ALLUSERSPROFILE=C:\ProgramData
> ANDROID_HOME=C:\ANDROID
> APPDATA=C:\Users\Shujahuddin\AppData\Roaming
> ChocolateyInstall=C:\ProgramData\chocolatey
> ChocolateyLastPathUpdate=132142359870397115
> CommonProgramFiles=C:\Program Files\Common Files
> CommonProgramFiles(x86)=C:\Program Files (x86)\Common Files
> CommonProgramW6432=C:\Program Files\Common Files
> COMPUTERNAME=SHUJA-LAPTOP
> ComSpec=C:\WINDOWS\system32\cmd.exe
> DriverData=C:\Windows\System32\Drivers\DriverData
> FLUTTER=C:\Flutter_v1.9.1_hotfix.2_stable\bin
> HOMEDRIVE=C:
> HOMEPATH=\Users\Shujahuddin
> JAVA_HOME=C:\Program Files\Java\jdk1.8.0_221
> LOCALAPPDATA=C:\Users\Shujahuddin\AppData\Local
> LocalLaravel=C:\laravel\vendor\bin
> LOGONSERVER=\\SHUJA-LAPTOP
> NUMBER_OF_PROCESSORS=4
> OneDrive=C:\Users\Shujahuddin\OneDrive
> OS=Windows_NT
> Path=C:\Program Files (x86)\Common Files\Oracle\Java\javapath;C:\WINDOWS\system32;C:\WINDOWS;C:\WINDOWS\System32\Wbem;C:\WINDOWS\System32\WindowsPowerShell\v1.0\;C:\WINDOWS\System32\OpenSSH\;C:\Program Files\Microsoft SQL Server\110\Tools\Binn\;C:\Program Files (x86)\Microsoft SQL Server\110\Tools\Binn\;C:\Program Files\Microsoft SQL Server\110\DTS\Binn\;C:\Program Files (x86)\Microsoft SQL Server\110\Tools\Binn\ManagementStudio\;C:\Program Files (x86)\Microsoft SQL Server\110\DTS\Binn\;C:\Program Files\Microsoft\Web Platform Installer\;C:\Program Files (x86)\Microsoft ASP.NET\ASP.NET Web Pages\v1.0\;C:\Program Files (x86)\Windows Kits\8.1\Windows Performance Toolkit\;C:\composer;C:\xampp\php;C:\laravel\vendor\bin;C:\Program Files\nodejs\;C:\Program Files\Git\cmd;C:\Flutter_v1.9.1_hotfix.2_stable\bin;C:\Program Files\Java\jdk1.8.0_221;C:\ANDROID\tools;C:\ANDROID\platform-tools;C:\ProgramData\chocolatey\bin;C:\Users\Shujahuddin\AppData\Local\Microsoft\WindowsApps;;C:\Users\Shujahuddin\AppData\Local\Programs\Microsoft VS Code\bin;C:\Users\Shujahuddin\AppData\Roaming\Composer\vendor\bin;C:\Users\Shujahuddin\AppData\Roaming\npm;C:\Users\Shujahuddin\AppData\Local\Programs\Fiddler;%USERPROFILE%\AppData\Local\Microsoft\WindowsApps
> PATHEXT=.COM;.EXE;.BAT;.CMD;.VBS;.VBE;.JS;.JSE;.WSF;.WSH;.MSC
> PHP=C:\xampp\php
> PROCESSOR_ARCHITECTURE=AMD64
> PROCESSOR_IDENTIFIER=Intel64 Family 6 Model 142 Stepping 9, GenuineIntel
> PROCESSOR_LEVEL=6
> PROCESSOR_REVISION=8e09
> ProgramData=C:\ProgramData
> ProgramFiles=C:\Program Files
> ProgramFiles(x86)=C:\Program Files (x86)
> ProgramW6432=C:\Program Files
> PROMPT=$P$G
> PSModulePath=C:\Program Files\WindowsPowerShell\Modules;C:\WINDOWS\system32\WindowsPowerShell\v1.0\Modules;C:\Program Files (x86)\Microsoft SQL Server\110\Tools\PowerShell\Modules\
> PUBLIC=C:\Users\Public
> SESSIONNAME=Console
> SystemDrive=C:
> SystemRoot=C:\WINDOWS
> TEMP=C:\Users\SHUJAH\~1\AppData\Local\Temp
> TMP=C:\Users\SHUJAH\~1\AppData\Local\Temp
> USERDOMAIN=SHUJA-LAPTOP
> USERDOMAIN_ROAMINGPROFILE=SHUJA-LAPTOP
> USERNAME=Shujahuddin
> USERPROFILE=C:\Users\Shujahuddin
> VS110COMNTOOLS=C:\Program Files (x86)\Microsoft Visual Studio 11.0\Common7\Tools\
> VS120COMNTOOLS=C:\Program Files (x86)\Microsoft Visual Studio 12.0\Common7\Tools\
> windir=C:\WINDOWS

**Comment by shajji1** [comment]
@Hixie, @escamoteur ...  any help

**Comment by shajji1** [comment]
@Hixie, @escamoteur i figure out that i'm facing this issue due to **KES** _Kaspersky Endoint Security_, when my network administrator disable it, all commands working smoothly, now the problem is we can't disable **KES** permenantly. is there any flag to trust those certificates via cli or i should close this issue, no offense. and one more thing if you suggest to close this issue and i did, does that mean new feature request is also closed.

**Comment by escamoteur** [comment]
Thanks for telling, we will look into it.

**Comment by oom-** [comment]
I have the same problem here but I'm at my home and not using any special software. I disabled all my protections and the problem still occurs.

```
Running "flutter pub upgrade" in testproject...
Got TLS error trying to find package path_provider at https://pub.dartlang.org.
pub upgrade failed (server unavailable) -- attempting retry 1 in 1 second...
Got TLS error trying to find package cupertino_icons at https://pub.dartlang.org.
pub upgrade failed (server unavailable) -- attempting retry 2 in 2 seconds...
Got TLS error trying to find package open_file at https://pub.dartlang.org.
pub upgrade failed (server unavailable) -- attempting retry 3 in 4 seconds...
Got TLS error trying to find package uuid at https://pub.dartlang.org.
pub upgrade failed (server unavailable) -- attempting retry 4 in 8 seconds...
Got TLS error trying to find package cupertino_icons at https://pub.dartlang.org.
pub upgrade failed (server unavailable) -- attempting retry 5 in 16 seconds...
Got TLS error trying to find package path_provider at https://pub.dartlang.org.
pub upgrade failed (server unavailable) -- attempting retry 6 in 32 seconds...
Running "flutter pub upgrade" in cov19genattest...              (This is taking an unexpectedly long time.)   
```

I figured out that I cannot at all reach the "pub.dartlang.org" website: 

- Neither on my phone: `DNS_PROBE_FINISHED_NXDOMAIN`
- Neither on my desktop: `ERR_CONNECTION_CLOSED` (after setting 1.1.1.1 as DNS server)

The tracert is fine and reach the target on the desktop, regarding this issue here: [https://github.com/dart-lang/pub-dev/issues/1767](https://github.com/dart-lang/pub-dev/issues/1767)

It looks like it's because my IP adress is in IPV6, then I try to disable it in my router + on my network interface (I'm on windows 10). Still not working.

Then I updated my DNS to the one of google 8.8.8.8 and then it works.
But in the end `pub.darlang.org` is redirecting me to `pub.dev`, what a mess...

![image](https://user-images.githubusercontent.com/11071671/97775412-abea4700-1b60-11eb-8ddc-fa934c16ae08.png)

**Comment by Paul74** [comment]
Same error here. Windows 10 in a corporate environment (so proxy/firewall...).
Running _flutter pub get -v_ I get:

HTTP error:
[   +1 ms]     | TlsException: Failed to find root cert file (OS Error)

note that if I browse to https://pub.flutter-io.cn/api/packages/packagename  I correctly get di json file.

So how to install that root cert file??

**Comment by MGNarva** [comment]
> Same error here. Windows 10 in a corporate environment (so proxy/firewall...).
> Running _flutter pub get -v_ I get:
> 
> HTTP error:
> [ +1 ms] | TlsException: Failed to find root cert file (OS Error)
> 
> note that if I browse to https://pub.flutter-io.cn/api/packages/packagename I correctly get the json file.
> 
> So how to install that root cert file??

Have you been able to solve the problem?
In my work we are with the same error

**Comment by Paul74** [comment]
> > Same error here. Windows 10 in a corporate environment (so proxy/firewall...).
> > Running _flutter pub get -v_ I get:
> > HTTP error:
> > [ +1 ms] | TlsException: Failed to find root cert file (OS Error)
> > note that if I browse to https://pub.flutter-io.cn/api/packages/packagename I correctly get the json file.
> > So how to install that root cert file??
> 
> Have you been able to solve the problem?
> In my work we are with the same error

nope. I gave up. I ended up by remotely connecting to another pc with direct internet access. works pretty well.

**Comment by MGNarva** [comment]
> > > Same error here. Windows 10 in a corporate environment (so proxy/firewall...).
> > > Running _flutter pub get -v_ I get:
> > > HTTP error:
> > > [ +1 ms] | TlsException: Failed to find root cert file (OS Error)
> > > note that if I browse to https://pub.flutter-io.cn/api/packages/packagename I correctly get the json file.
> > > So how to install that root cert file??
> > 
> > 
> > Have you been able to solve the problem?
> > In my work we are with the same error
> 
> nope. I gave up. I ended up by remotely connecting to another pc with direct internet access. works pretty well.

Paul I'm sorry you didn't get another answer.
In my job the security engineering sector added the ssl exlusion 'https://pub.dartlang.org/' for Flutter. After the change it was executed correctly.

**Comment by Paul74** [comment]
thanks for that. 

 > Paul I'm sorry you didn't get another answer.
> In my job the security engineering sector added the ssl exlusion 'https://pub.dartlang.org/' for Flutter. After the change it was executed correctly.

**Comment by matanlurey** [comment]
This feature is not planned.
